cli/factory.rs

```diff
@@ -537,7 +539,19 @@ impl CliFactory {
   }
 
   pub fn graph_container(&self) -> &Arc<ModuleGraphContainer> {
-    self.services.graph_container.get_or_init(Default::default)
+    self.services.graph_container.get_or_init(|| {
+      let graph_kind = match self.options.sub_command() {
+        DenoSubcommand::Cache(_) => GraphKind::All,
```

Comment:
What about `deno info`?

Reply:
This is only for the graph container, which is used with `deno cache` for some reason, but not deno info. Probably `deno cache` shouldn't use this as the graph container is only used for stuff like deno run when the graph size can expand due to workers or discovered dynamic imports. I will open a new PR.

https://github.com/denoland/deno/pull/19392/files#diff-173ccedeb7f83e3e957f25e701e7fd898850183300d22f07698f9b37d7bdbbd5R47

Reply:
Oh, ok it goes through this for `deno cache` because it calls `prepare_module_load`, which does a bunch of stuff like also write to the lockfile. I think it's too much work to bother separating it out atm.

Reply:
I added a todo comment here: https://github.com/denoland/deno/pull/19393